src/test/java/org/gridsuite/voltageinit/service/ReportServiceTest.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.voltageinit.service;
+
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.commons.reporter.ReporterModel;
+import okhttp3.HttpUrl;
+import okhttp3.mockwebserver.Dispatcher;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.gridsuite.voltageinit.server.service.ReportService;
+import org.gridsuite.voltageinit.utils.ContextConfigurationWithTestChannel;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.client.RestClientException;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+/**
+ * @author Anis TOURI <anis.touri at rte-france.com>
+ */
+@SpringBootTest
+@RunWith(SpringRunner.class)
+@ContextConfigurationWithTestChannel
+public class ReportServiceTest {
+
+    private static final UUID REPORT_UUID = UUID.fromString("7928181c-7977-4592-ba19-88027e4254e4");
+    private static final UUID REPORT_ERROR_UUID = UUID.fromString("9928181c-7977-4592-ba19-88027e4254e4");
+
+    private static final String REPORT_JSON = "{\"version\":\"1.0\",\"reportTree\":{\"taskKey\":\"test\"},\"dics\":{\"default\":{\"test\":\"a test\"}}}";
+
+    private MockWebServer server;
+
+    @Autowired
+    private ReportService reportService;
+
+    @Before
+    public void setUp() throws IOException {
+        String mockServerUri = initMockWebServer();
+        reportService.setReportServiceBaseUri(mockServerUri);
+    }
+
+    @After
+    public void tearDown() {
+        try {
+            server.shutdown();
+        } catch (Exception e) {
+            // Nothing to do
+        }
+    }
+
+    private String initMockWebServer() throws IOException {
```

Comment:
As discussed with the techleads , the new test classes should use WireMockServer to avoid the ugly dispatch.
But if it's time consuming to change it don't bother.

Reply:
Done